vllm/model_executor/layers/linear.py

```diff
@@ -23,8 +23,8 @@ class LinearMethodBase(ABC):
     @abstractmethod
     def create_weights(self, input_size_per_partition: int,
                        output_size_per_partition: int, input_size: int,
-                       output_size: int,
-                       params_dtype: torch.dtype) -> Dict[str, Any]:
+                       output_size: int, params_dtype: torch.dtype,
+                       output_partition_sizes: List[int]) -> Dict[str, Any]:
```

Comment:
I wasn't thrilled about having to pass `output_partition_sizes` in, but we need that information to get our job done, and it seemed like a minor addition given that we already pass in other more detailed information about sizes.  Conceivably we could pass that in INSTEAD of `output_size_per_partition` because the former sums up to the latter, but I was hesitant to change that much other code without the approval of the vllm guys.  Happy to take opinions.

Reply:
Actually I made it optional elsewhere.  I think create_weights is good as is. follows the pattern.

Reply:
It seems like `output_partition_sizes` is a more generalized version of `output_size_per_partition`?  Would it make sense to replace `output_size_per_partition` with `output_partition_sizes` rather than having both?

Reply:
That's a thought, and you are right, but it's a little scary to do any refactor in Python because there's no compiler to flag where you need to change a signature, and no way to guarantee that the runtime goes through all the paths to make sure you didn't break anything.  I consider this a huge glaring problem with Python...

Reply:
That said maybe I'll try it.   In this case since it's not in a constructor, it's a bit easier to guarantee you got them all.  It's the __init__ calls that are tricky to change.